### PR TITLE
Support arithmetic binary in Delta check constraints

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Support arithmetic binary in Delta check constraints

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake
* Add support for arithmetic binary expressions in check constraints. ({issue}`16721`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/expression/TestSparkExpressions.java [hunk 1]
```diff
@@ -127,10 +127,22 @@ public void testIdentifier()
...
+        assertExpressionTranslates("a = b / 1", "(\"a\" = (\"b\" / 1))");
+        assertExpressionTranslates("a = b & 1", "(\"a\" = (bitwise_and(\"b\", 1)))");
+        assertExpressionTranslates("a = b ^ 1", "(\"a\" = (bitwise_xor(\"b\", 1)))");
+    }
+
     @Test
     public void testInvalidNotBoolean()
     {
-        assertParseFailure("a + a");
+        assertParseFailure("'Spark' || 'SQL'");
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/expression/TestSparkExpressions.java:145]: Is this different than what we have on Line: 173?

```java
       assertParseFailure("a = 'Spark' || 'SQL'");
```

Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/expression/TestSparkExpressions.java:145]: Different. `'Spark' || 'SQL'` returns string. `a = 'Spark' || 'SQL'` returns boolean.

**Review by krvikash — APPROVED** [review]

**Comment by ebyhr** [comment]
CI hit #11131

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/ArithmeticBinaryExpression.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    private final Operator operator;
+    private final SparkExpression left;
+    private final SparkExpression right;
+
+    public ArithmeticBinaryExpression(Operator operator, SparkExpression left, SparkExpression right)
+    {
+        this.operator = operator;
+        this.left = left;
+        this.right = right;
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/ArithmeticBinaryExpression.java:53]: `rnn`

**Review by findepi — APPROVED** [review]
Please add parser tests showing that expression parser honors arithmetic operators priorities.

**Pull request merged**